app/controllers/concerns/find_common.rb

```diff
@@ -54,4 +54,8 @@ def resource_class_for(resource)
       return nil
     end
   end
+
+  def find_hosts_count
+    @hosts_count = Host::Managed.reorder('').authorized.group("#{resource_name}_id").count
+  end
```

Comment:
Thanks for bringing this up. My first observation is explicit vs implicit. Do we really need to fetch the count for each individual index page? Are you sure all index pages use it? I'd rather prefer a helper method `resource_count` that would cache the number for the lifetime of request. Meaning fetching it only when it's really needed.